crates/volta-core/src/path/unix.rs

```diff
@@ -109,12 +110,34 @@ pub fn shim_file(toolname: &str) -> Fallible<PathBuf> {
     Ok(shim_dir()?.join(toolname))
 }
 
+// this is not currently used by anything
 pub fn volta_file() -> Fallible<PathBuf> {
     Ok(volta_home()?.join("volta"))
 }
 
+// check that it exists - if not, check some other locations
 pub fn shim_executable() -> Fallible<PathBuf> {
-    Ok(volta_home()?.join("shim"))
+    // if VOLTA_SHIM is set, try that first
+    // (not documented yet, as it's currently only used for testing)
+    if let Some(shim_location) = env::var_os("VOLTA_SHIM") {
+        return Ok(Path::new(&shim_location).to_path_buf());
```

Comment:
`PathBuf` implements `From<OsString>`, so this can be simplified with `.into()`:

```suggestion
        return Ok(shim_location.into());
```

Reply:
ah, awesome